### DISCUSSION TITLE
next.js status code 200 on 404 page

### DISCUSSION THREAD

**Discussion by kfi** [vercel/vercel#4723] [Help]
Hi,

locally with "next" and "serve out" (for the static export) my 404 page correctly returns a 404 http status code, but on vercel it returns 200. I use a 404.js file to customize the 404 page.

Can someone point me into the right direction?

Thank you.

**Comment by styfle** [comment]
Hi @kfi 

Next.js will use the custom 404 page located in [`pages/404.js`](https://nextjs.org/docs/advanced-features/custom-error-page).

You will need to use the default Project Settings such that `next build` is called somewhere in your [Build Command](https://vercel.com/docs/v2/build-step).

**Comment by pacocoursey** [comment]
@kfi Would you post a deployment URL or sample source code?

Are you using `next export` or automatic static optimization?

Is your `pages/404.js` file static or using `getInitialProps`?

**Reply by kfi** [reply]
I used the automatic static optimization. All pages use `getStaticProps` and the automatic static optimization correctly shows that all pages use SSG. My custom app uses `getInitialProps` to fetch static data for header and footer, which might be the issue. But as I understand, this should not affect the static generation of my pages because I use exclusively `getStaticProps`.

I changed the deployment to `next export` and now the 404 pages send the correct status code. It looks like the serving just works differently with static files.

**Comment by fab1an** [comment]
This is a bug: This is a bug: https://github.com/vercel/next.js/issues/43831
